FHAC-760: Fix for most of the critical Sonar bugs.  These bugs mostly…

danielrf123 opened the pull request:
… were related to the use of static and non-serializable variables.

Commit:
FHAC-760: Fix for most of the critical Sonar bugs.  These bugs mostly were related to the use of static and non-serializable variables.

christophermay07 commented:
add to whitelist

christophermay07 commented on Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdb/model/Patient.java:
```diff
@@ -40,7 +40,9 @@
-    /**
+	private static final long serialVersionUID = -5941897049106149743L;
+
+	/**
```
Make sure you have tabs to spaces, and 4 spaces per indent

christophermay07 replied:
Tab to space issue still exists here.

christophermay07 commented on Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/callback/openSAML/CertificateManagerImplTest.java:
```diff
@@ -52,12 +52,12 @@
         final HashMap<String, String> trustStoreMap = new HashMap<>();
         trustStoreMap.put(CertificateManagerImpl.TRUST_STORE_KEY, TRUST_STORE_PATH);
-        trustStoreMap.put(CertificateManagerImpl.TRUST_STORE_PASSWORD_KEY, "changeit");
+        trustStoreMap.put(CertificateManagerImpl.TRUST_STORE_PSWRD_KEY, "changeit");
```
Please change these back to password, it's still the most descriptive name for the constant.

christophermay07 commented on Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7PRPA201301Transforms.java:
```diff
@@ -545,7 +545,8 @@ private static MCCIMT000100UV01Sender copySender(MCCIMT000300UV01Sender orig) {
     }
 
-    protected void setHL7MessageIdGenerator(HL7MessageIdGenerator idGenerator) {
+    @SuppressWarnings("static-access")
```
A suppressed warning needs to be documented first -- we can discuss tomorrow.

christophermay07 commented on Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/CXFPasswordCallbackHandler.java:
```diff
@@ -72,7 +72,7 @@ public final void handle(final Callback[] callbacks) throws IOException, Unsuppo
         if (keystorePassword == null || keystorePassword.isEmpty()) {
             throw new IOException("keystore password was not provided. Please provide the system property: "
-                    + SYSTEM_PROPERTY_KEYSTORE_PASSWORD);
+                    + SYSTEM_PROPERTY_KEYSTORE_PSWRD);
```
Same as other comment, change back to password.

Commit:
New commit for FHAC-760.  This has the updates for the comments provided in the original pull request.

Commit:
Removed the static logStatus field from AuditRepositoryOrchImpl.java.  Undid the renaming of the password field in CXFPasswordCallbackHandler.

Commit:
Switched back the method headers in AuditRepositoryOrchImpl that were changed to static.

christophermay07 commented on Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoginBean.java:
```diff
@@ -72,7 +72,7 @@
     public Boolean isCorrect = false;
 
-    private static boolean firstTimeLogged = true;
+    private boolean firstTimeLogged = true;
```
I haven't looked into the code yet, but I think this was static to lazy initialize something when the first user logs in, but never again after that.  Make sure this change is not going to cause side effects.

danielrf123 replied:
Another option is to make checkDisplays() static.  If we go with that we would have to to change login and loginAndNavigate to static, to pass the next Sonar scan. I don't know if that would bring up a problem since loginAndNavigate() is called from login.xhtml.

christophermay07 replied:
Can you move the associated logic to happen once per application (during initialization)?

christophermay07 commented on Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/RoleService.java:
```diff
@@ -29,13 +29,15 @@
  * @author jasonasmith
  */
-public interface RoleService {
+public interface RoleService extends Serializable{
```
Add a space before the "{" (does the auto-formatter not take care of this?)

danielrf123 replied:
I committed this before I restarted eclipse after making the formatter preferences update.  I will make the change manually for the next commit.

christophermay07 commented on Product/Production/Adapters/General/CONNECTConsumerPreferencesProfileGUI/src/main/java/gov/hhs/fha/nhinc/adapter/cppgui/valueobject/PatientVO.java:
```diff
@@ -26,12 +26,18 @@
+
+import java.io.Serializable;
+
+import gov.hhs.fha.nhinc.adapter.cppgui.valueobject.PatientPreferencesVO;
```
Again, check auto-formatting settings.  There should be a single line between packages, no lines between imports, and imports should be organized alphabetically.

danielrf123 replied:
I agree.

christophermay07 commented on Product/Production/Adapters/General/CONNECTConsumerPreferencesProfileGUI/src/main/java/gov/hhs/fha/nhinc/adapter/cppgui/valueobject/PatientVO.java:
```diff
@@ -26,12 +26,18 @@
+public class PatientVO implements Serializable{
+
+	private static final long serialVersionUID = 1L;
+	private String firstName;
```
Same comment about the curly brace, also run tabs to spaces here.

danielrf123 replied:
I agree.

christophermay07 commented on Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/service/ConfigurationFault.java:
```diff
@@ -59,7 +59,7 @@
     private String message = "";
 
-    private ConfigurationError error = ConfigurationError.Unknown;
+    private transient ConfigurationError error = ConfigurationError.Unknown;
```
Question: Do we not need this information to be serialized?

danielrf123 replied:
Can't find the SONAR defect that mentioned why that variable should have been made transient.  Will update when I find it/remember.

christophermay07 replied:
Something I completely forgot:  All enums are serializable:

http://docs.oracle.com/javase/7/docs/api/java/lang/Enum.html

christophermay07 commented on Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/service/ConfigurationServiceException.java:
```diff
@@ -55,7 +55,7 @@
 
     @SuppressWarnings("unused")
-    private ConfigurationFault fault = new ConfigurationFault();
+    private transient ConfigurationFault fault = new ConfigurationFault();
```
Can this variable be removed altogether?

danielrf123 replied:
It can and will be removed.  There aren't any references of it outside that class.
The constructor around line 108 has a ConfigurationFault object as a parameter.  Can that constructor be removed?

christophermay07 commented on Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/asyncmsgs/model/AsyncMsgRecord.java:
```diff
@@ -33,7 +34,7 @@
  * @author JHOPPESC, richard.ettema
  */
-public class AsyncMsgRecord {
+public class AsyncMsgRecord implements Serializable{
```
Same spacing issue as before.

danielrf123 replied:
Agree.

christophermay07 commented on Product/Production/Adapters/General/CONNECTConsumerPreferencesProfileGUI/src/main/java/gov/hhs/fha/nhinc/adapter/cppgui/valueobject/PatientVO.java:
```diff
@@ -26,12 +26,18 @@
-    private String firstName;
+public class PatientVO implements Serializable{
+
+	private static final long serialVersionUID = 1L;
```
You created a unique uid in one of the classes below, but used 1L here; any reason for the mix and match?

danielrf123 replied:
No reason.  I'll switch this to a generated ID to be consistent.

Commit:
FHAC-760 Updates from PR comments.

Commit:
FHAC-760: Reformatted files so extra tabbing isn't shown.

mhpnguyen commented on Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7PRPA201306Transforms.java:
```diff
@@ -203,9 +209,9 @@ public PRPAIN201306UV02 createPRPA201306ForErrors(PRPAIN201305UV02 oRequest, Str
         try {
             LOG.info("Attempting to retrieve property: " + NhincConstants.HOME_COMMUNITY_ID_PROPERTY
-                + " from property file: " + NhincConstants.GATEWAY_PROPERTY_FILE);
+                    + " from property file: " + NhincConstants.GATEWAY_PROPERTY_FILE);
```
Should we use parameterized logging such as LOG.info("test{}" ,variable) to improve performance?

danielrf123 replied:
That might be good.  Except we should ignore this change because line wrapping was done from the auto-formatting in eclipse.  I didn't intentionally make that change.

mhpnguyen commented on Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7PRPA201306Transforms.java:
```diff
@@ -678,7 +684,7 @@ protected boolean areSenderFieldsNull(PRPAIN201305UV02 oRequest) {
         if (NullChecker.isNullish(oRequest.getSender().getDevice().getId().get(0).getRoot())) {
             LOG.error("The device id value (i.e. II.getRoot() or "
-                + "oRequest.getSender().getDevice().getId().get(0).getRoot()) is null.");
+                    + "oRequest.getSender().getDevice().getId().get(0).getRoot()) is null.");
```
Please use parameterized logging

danielrf123 replied:
That update in that logging was done from the auto-formatting.  It had nothing to do with the change done for FHAC-760.

christophermay07 commented on Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/RoleService.java:
```diff
@@ -26,19 +26,22 @@
  * @author jasonasmith
  */
-public interface RoleService {
+public interface RoleService extends Serializable {
```
Does every class that implements this, also define serial uid?

danielrf123 replied:
One class did and it now has one defined.

christophermay07 commented on Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/service/ConfigurationServiceException.java:
```diff
@@ -76,39 +72,27 @@ public ConfigurationServiceException(String message) {
-     *            A ConfigurationFault.
+     * @param aFault A ConfigurationFault.
      */
     public ConfigurationServiceException(ConfigurationFault aFault) {
```
This appears to not be used anywhere, can we remove it?

danielrf123 replied:
It can be removed.

Commit:
Removed checkDisplay() from LoginBean.java and moved it to a servlet on startup.

RoleServiceImpl needs to have a serialized ID.

Corrected file error in ConnectValidation-soapui-project.xml.  Reformatted modified files so extra white space is removed.

christophermay07 commented on Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hss/fha/nhinc/admingui/servlet/InitServlet.java:
```diff
@@ -0,0 +1,57 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hss.fha.nhinc.admingui.servlet;
```
Wrong package space (should be hhs); make sure to update in web.xml as well.

christophermay07 commented on Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hss/fha/nhinc/admingui/servlet/InitServlet.java:
```diff
@@ -0,0 +1,57 @@
+public class InitServlet extends HttpServlet {
+
+	private static final long serialVersionUID = -7194100487044821581L;
+	private static final Logger LOG = LoggerFactory.getLogger(InitServlet.class);
```
If we're not logging anything, remove this and the associated imports.

christophermay07 commented on Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hss/fha/nhinc/admingui/servlet/InitServlet.java:
```diff
@@ -0,0 +1,57 @@
+    public void init(ServletConfig config) throws ServletException {
+		super.init(config);
+		
+		//The below checkDisplay() calls were originally called in a checkDisplay method in the LoginBean.
```
We can find this information out via commit history -- it would be better to have comments in the code to explain why it's there, instead of why it's not somewhere else.

christophermay07 commented on Product/Production/Common/CONNECTCoreLib/pom.xml:
```diff
@@ -28,7 +28,7 @@
             <artifactId>CONNECTCommonTypesLib</artifactId>
         </dependency>
-
+        
```
When modifying an XML, make sure to do two things - trim trailing space, and pretty print formatting.

Commit:
Peer review corrections made.

christophermay07 commented:
I've reviewed this PR, endorsing for merge without further comment.

mhpnguyen commented on Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7PRPA201306Transforms.java:
```diff
@@ -126,7 +130,7 @@ public PRPAIN201306UV02 createPRPA201306ForPatientNotFound(PRPAIN201305UV02 oReq
         if (bRequiredFieldsAreNull) {
             LOG.error("One or more required fields from the patient discovery request "
-                + "for the patient not found scenario are null.");
+                    + "for the patient not found scenario are null.");
```
since it is expensive to concat String using + operation, should we use StringBuilder to build if the error message is long?

danielrf123 replied:
I agree, but I did not touch this code.  The eclipse formatter made the change since I saved the file.  So I think I should leave it alone for my commits for FHAC-760?

mhpnguyen commented:
Peer review PR and endorse for merge without any further comments
